src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1375,13 +1382,28 @@ void Connection::receive_pong(milliseconds_type timestamp)
         m_client.m_roundtrip_time_handler(m_previous_ping_rtt); // Throws
 }
 
+bool Connection::check_session_history(session_ident_type session_ident)
```

Comment:
This is to prevent any pending messages in the websocket from closing the connection if they come in after the session has been closed. The other part of the fix is to not allow Session to call functions on the SessionWrapper reference if the session is currently deactivated or being deactivated